src/handlers/permissions-grant.ts

```diff
@@ -29,22 +30,57 @@ export class PermissionsGrantHandler implements MethodHandler {
       return messageReplyFromError(e, 401);
     }
 
-    const { scope, conditions, ...propertiesToIndex } = message.descriptor;
-    const indexes: { [key: string]: string } = {
-      author: permissionsGrant.author!,
-      ...propertiesToIndex,
-    };
+    const indexes = PermissionsGrantHandler.constructIndexes(permissionsGrant);
 
     // If we have not seen this message before, store it
     const messageCid = await Message.getCid(message);
     const existingMessage = await this.messageStore.get(tenant, messageCid);
     if (existingMessage === undefined) {
       await this.messageStore.put(tenant, message, indexes);
-      await this.eventLog.append(tenant, messageCid);
+      const additionalIndexes = PermissionsGrantHandler.constructAdditionalIndexes(permissionsGrant);
+      await this.eventLog.append(tenant, messageCid, { ...indexes, ...additionalIndexes });
     }
 
     return {
       status: { code: 202, detail: 'Accepted' }
     };
   }
+
+  /**
+  * Indexed properties needed for MessageStore indexing.
+  */
+  static constructIndexes(
+    permissionsGrant: PermissionsGrant,
+  ): Record<string, string> {
+    const message = permissionsGrant.message;
+    const { scope, conditions, ...propertiesToIndex } = message.descriptor;
+
+    const indexes: Record<string, any> = {
+      author: permissionsGrant.author!,
+      ...propertiesToIndex,
+    };
+    return indexes;
+  }
+
+  /**
+   * Additional indexes that are not needed within the MessageStore but are necessary within the EventLog.
+   */
+  static constructAdditionalIndexes(
```

Comment:
nit: rename to `constructIndexes` -> `constructMessageStoreIndexes` and `constructAdditionalIndexes` -> `constructAdditionalEventLogIndexes`.